macOS core build failing

### Bug

The [clp-core-build-macos](https://github.com/y-scope/clp/blob/main/.github/workflows/clp-core-build-macos.yaml) workflow is failing seemingly due to:

```
In file included from /Users/runner/work/clp/clp/components/core/src/clp/string_utils/../spdlog_with_specializations.hpp:4:
In file included from /usr/local/include/fmt/format.h:41:
/usr/local/include/fmt/base.h:1402:23: error: no matching member function for call to 'format'
    ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
                   ~~~^~~~~~
/usr/local/include/fmt/base.h:1383:21: note: in instantiation of function template specialization 'fmt::detail::value<fmt::context>::format_custom_arg<clp::ErrorCode, fmt::formatter<clp::ErrorCode>>' requested here
    custom.format = format_custom_arg<
                    ^
/usr/local/include/spdlog/logger.h:80:9: note: in instantiation of function template specialization 'spdlog::logger::log_<clp::ErrorCode &>' requested here
        log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
        ^
/Users/runner/work/clp/clp/components/core/src/clp/clp/FileDecompressor.cpp:21:13: note: in instantiation of function template specialization 'spdlog::logger::log<clp::ErrorCode &>' requested here
            SPDLOG_ERROR("Failed to open encoded file, error_code={}", error_code);
            ^
/usr/local/include/spdlog/spdlog.h:333:31: note: expanded from macro 'SPDLOG_ERROR'
    #define SPDLOG_ERROR(...) SPDLOG_LOGGER_ERROR(spdlog::default_logger_raw(), __VA_ARGS__)
                              ^
/usr/local/include/spdlog/spdlog.h:332:9: note: expanded from macro 'SPDLOG_LOGGER_ERROR'
        SPDLOG_LOGGER_CALL(logger, spdlog::level::err, __VA_ARGS__)
        ^
/usr/local/include/spdlog/spdlog.h:288:19: note: expanded from macro 'SPDLOG_LOGGER_CALL'
        (logger)->log(spdlog::source_loc{__FILE__, __LINE__, SPDLOG_FUNCTION}, level, __VA_ARGS__)
                  ^
/Users/runner/work/clp/clp/components/core/src/clp/string_utils/../spdlog_with_specializations.hpp:19:10: note: candidate function template not viable: 'this' argument has type 'const fmt::formatter<clp::ErrorCode>', but method is not marked const
    auto format(clp::ErrorCode const& error_code, FormatContext& ctx) {
         ^
1 warning and 1 error generated.
```

This is likely because the `fmt` version available from Homebrew changed from 10.2.1.1 to 11.0.2.

### CLP version

ea7678fca67b6f560667dda9f7ac7b7e723b6ff7

### Environment

* macOS 12
* fmt 11.0.2

### Reproduction steps

Run the workflow.

--- a/components/core/src/clp/spdlog_with_specializations.hpp
+++ b/components/core/src/clp/spdlog_with_specializations.hpp
@@ -16,7 +16,7 @@ struct fmt::formatter<clp::ErrorCode> {
     }
 
     template <typename FormatContext>
-    auto format(clp::ErrorCode const& error_code, FormatContext& ctx) {
+    auto format(clp::ErrorCode const& error_code, FormatContext& ctx) const {
         return fmt::format_to(ctx.out(), "{}", static_cast<size_t>(error_code));
     }
 };
@@ -30,7 +30,8 @@ struct fmt::formatter<clp::ffi::search::ExactVariableToken<encoded_variable_t>> 
 
     template <typename FormatContext>
     auto
-    format(clp::ffi::search::ExactVariableToken<encoded_variable_t> const& v, FormatContext& ctx) {
+    format(clp::ffi::search::ExactVariableToken<encoded_variable_t> const& v,
+           FormatContext& ctx) const {
         return fmt::format_to(
                 ctx.out(),
                 "ExactVariableToken(\"{}\") as {}",
@@ -48,7 +49,8 @@ struct fmt::formatter<clp::ffi::search::WildcardToken<encoded_variable_t>> {
     }
 
     template <typename FormatContext>
-    auto format(clp::ffi::search::WildcardToken<encoded_variable_t> const& v, FormatContext& ctx) {
+    auto
+    format(clp::ffi::search::WildcardToken<encoded_variable_t> const& v, FormatContext& ctx) const {
         return fmt::format_to(
                 ctx.out(),
                 "WildcardToken(\"{}\") as {}TokenType({}){}",

--- a/components/core/src/glt/spdlog_with_specializations.hpp
+++ b/components/core/src/glt/spdlog_with_specializations.hpp
@@ -16,7 +16,7 @@ struct fmt::formatter<glt::ErrorCode> {
     }
 
     template <typename FormatContext>
-    auto format(glt::ErrorCode const& error_code, FormatContext& ctx) {
+    auto format(glt::ErrorCode const& error_code, FormatContext& ctx) const {
         return fmt::format_to(ctx.out(), "{}", static_cast<size_t>(error_code));
     }
 };
@@ -30,7 +30,8 @@ struct fmt::formatter<glt::ffi::search::ExactVariableToken<encoded_variable_t>> 
 
     template <typename FormatContext>
     auto
-    format(glt::ffi::search::ExactVariableToken<encoded_variable_t> const& v, FormatContext& ctx) {
+    format(glt::ffi::search::ExactVariableToken<encoded_variable_t> const& v,
+           FormatContext& ctx) const {
         return fmt::format_to(
                 ctx.out(),
                 "ExactVariableToken(\"{}\") as {}",
@@ -48,7 +49,8 @@ struct fmt::formatter<glt::ffi::search::WildcardToken<encoded_variable_t>> {
     }
 
     template <typename FormatContext>
-    auto format(glt::ffi::search::WildcardToken<encoded_variable_t> const& v, FormatContext& ctx) {
+    auto
+    format(glt::ffi::search::WildcardToken<encoded_variable_t> const& v, FormatContext& ctx) const {
         return fmt::format_to(
                 ctx.out(),
                 "WildcardToken(\"{}\") as {}TokenType({}){}",

